### PR TITLE
IPP for OpenCV 2017u2 initial enabling patch;

### PR DESCRIPTION
**Pull request by lupustr3**
### This pullrequest changes

Adds IPP for OpenCV 2017u2 initial integration.
Also contains redesigned IPP integrations for functions which are not compatible with new ICV package:
cv::Sobel
cv::Scharr
cv::Laplacian
cv::resize
cv::sort
cv::sortIdx
cv::calcHist
cv::erode
cv::dilate
cv::GaussianBlur
cv::medianBlur
cv::matchTemplate

### CODE REVIEW

**Comment by lupustr3** [comment]
Experimental, do not merge!

**Comment by lupustr3** [comment]
Now it works and can be merged.

Any more reviews, additional tests?

File: modules/core/include/opencv2/core/private.hpp [hunk 1]
```diff
@@ -230,44 +294,193 @@ static inline IppiPoint ippiPoint(int x, int y)
...
+
+        return IPP_MAX(1, (IPP_MIN((int)(opMemory/(l2cache*0.6)), threads)));
+    }
+    return 1;
+}
+#endif
+
+static inline int ippiSuggestThreadsNum(const cv::Mat &image, double multiplier)
+{
+    int threads = cv::getNumberOfCPUs();
```
Comment (alalek) [inline modules/core/include/opencv2/core/private.hpp]: Probably this should be consistent with implementation above: `cv::getNumThreads()` vs `cv::getNumberOfCPUs()`?

Comment (lupustr3) [inline modules/core/include/opencv2/core/private.hpp]: Right

**Comment by lupustr3** [comment]
new: matchTemplate IPP cutoff for templates bigger than half of image

**Comment by alalek** [comment]
Thank you! :+1:

**Pull request merged**